jerry-core/ecma/builtin-objects/ecma-builtin-array-prototype.cpp

```diff
@@ -644,61 +643,38 @@ ecma_builtin_array_prototype_object_push (ecma_value_t this_arg, /**< this argum
   /* 3. */
   ECMA_OP_TO_NUMBER_TRY_CATCH (length_var, length_value, ret_value);
 
-  uint32_t n = ecma_number_to_uint32 (length_var);
+  ecma_number_t n = ecma_uint32_to_number (ecma_number_to_uint32 (length_var));
 
   /* 5. */
   for (uint32_t index = 0;
-       index < arguments_number;
+       index < arguments_number && ecma_is_completion_value_empty (ret_value);
        index++, n++)
   {
     /* 5.a */
     ecma_value_t e_value = argument_list_p[index];
 
     /* 5.b */
-    ecma_string_t *n_str_p = ecma_new_ecma_string_from_uint32 (n);
+    ecma_string_t *n_str_p = ecma_new_ecma_string_from_number (n);
```

Comment:
It is better to use `ecma_new_ecma_string_from_uint32` instead of `ecma_new_ecma_string_from_number`, as it reduces memory usage.